include/flatbuffers/flexbuffers.h

```diff
@@ -377,12 +379,44 @@ class Reference {
       case TYPE_NULL: return 0;
       case TYPE_STRING: return flatbuffers::StringToInt(AsString().c_str());
       case TYPE_VECTOR: return static_cast<int64_t>(AsVector().size());
+      case TYPE_BOOL: return static_cast<int64_t>(AsBool());
       default:
       // Convert other things to int.
       return 0;
     }
   }
 
+  bool AsBool() const {
+    if (type_ == TYPE_BOOL) {
```

Comment:
The only issue I see with that you can lose semantic meaning.
A very contrived example.

```json
{ 
  "question" : "Is flex awesome", 
  "answer"   : "true", 
  "choices"  : ["true", "false", "unsure"]
}
```
Someone would expect the type of answer to always be a string by converting a string during parsing to a boolean you that meaning.

So which direction would you want me to go?